src/action/platform_info.rs

```diff
@@ -0,0 +1,394 @@
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+//! A handler and associated types for the platform info action.
+
+use crate::action::insttime::get_install_time;
+#[cfg(target_os = "windows")]
+use crate::fs::windows::dll_module_fileinfo;
+use crate::session::{self, Session};
+#[cfg(target_os = "macos")]
+use cocoa::{
+    appkit::*,
+    base::nil,
+    foundation::{NSInteger, NSProcessInfo},
+};
+use log::error;
+#[cfg(target_os = "macos")]
+use objc::{msg_send, sel, sel_impl};
+use std::env::consts::ARCH;
+#[cfg(target_family = "unix")]
+use std::ffi::CStr;
+use std::ffi::OsString;
+#[cfg(target_family = "unix")]
+use std::io::Error;
+#[cfg(target_family = "unix")]
+use std::mem;
+#[cfg(target_family = "unix")]
+use std::os::unix::prelude::OsStringExt as _;
+#[cfg(target_os = "windows")]
+use std::os::windows::ffi::OsStringExt as _;
+#[cfg(target_family = "unix")]
+use std::ptr;
+use std::time::{SystemTime, UNIX_EPOCH};
+#[cfg(target_os = "windows")]
+use windows::{
+    core::{Error, PWSTR},
+    w,
+    Win32::System::{
+        Diagnostics::Debug::{
+            PROCESSOR_ARCHITECTURE, PROCESSOR_ARCHITECTURE_AMD64, PROCESSOR_ARCHITECTURE_ARM,
+            PROCESSOR_ARCHITECTURE_IA64, PROCESSOR_ARCHITECTURE_INTEL,
+        },
+        SystemInformation::{
+            ComputerNamePhysicalDnsFullyQualified, ComputerNamePhysicalDnsHostname,
+            GetComputerNameExW, GetNativeSystemInfo, SYSTEM_INFO,
+        },
+    },
+};
+
+#[cfg(target_os = "linux")]
+#[link(name = "libc_version")]
+extern "C" {
+    fn libc_version() -> *const libc::c_char;
+}
+
+#[cfg(target_os = "macos")]
+#[allow(dead_code)]
+#[repr(C)]
+struct NSOperatingSystemVersion {
+    major_version: NSInteger,
+    minor_version: NSInteger,
+    patch_version: NSInteger,
+}
+
+/// A response type for the platform information action.
+struct Response {
+    /// The platform information.
+    platform_info: PlatformInfo,
+}
+
+/// The platform information.
+#[derive(Clone, Debug)]
+struct PlatformInfo {
+    /// The system platform (Windows|Darwin|Linux).
+    system: String,
+    /// The hostname of this system.
+    node: OsString,
+    /// The OS release identifier e.g. 7, OSX, debian.
+    release: String,
+    /// The OS version ID e.g. 6.1.7601SP1, 10.9.2, 14.04.
+    version: Option<String>,
+    /// The system architecture e.g. AMD64, x86_64.
+    machine: Option<String>,
+    // The kernel version string e.g. 6.1.7601, 13.1.0, 3.15-rc2.
+    kernel: String,
+    /// The system's fully qualified domain name.
+    fqdn: OsString,
+    /// When system was installed.
+    install_date: Option<SystemTime>,
+    /// The C library version.
+    libc_ver: Option<String>,
+    /// The architecture of this binary. (Note this can be different from
+    /// the machine architecture in the case of a 32 bit binary running
+    /// on a 64 bit system)
+    architecture: String,
```

Comment:
Since this just uses `std::env::consts::ARCH` this can be typed as `&'static str`.